### PR TITLE
[docs] Add doc for EAS SSO

### PR DESCRIPTION
**Pull request by wschurman**
# Why

This adds basic documentation for EAS Single Sign-On (SSO).

Closes ENG-9403

# How

Add doc

# Test Plan

start locally, visit http://localhost:3002/accounts/sso/.

# Checklist

- [ ] Documentation is up to date to reflect these changes (eg: https://docs.expo.dev and README.md).
- [ ] Conforms with the [Documentation Writing Style Guide](https://github.com/expo/expo/blob/main/guides/Expo%20Documentation%20Writing%20Style%20Guide.md)
- [ ] This diff will work correctly for `npx expo prebuild` & EAS Build (eg: updated a module plugin).

### CODE REVIEW

File: docs/pages/accounts/sso.mdx [hunk 1]
```diff
@@ -0,0 +1,62 @@
+---
+title: Single Sign-On (SSO)
+description: Learn about how your enterprise organization can use your Identity Provider to manage Expo users.
+---
+
+## Configure SSO for your organization
```
Comment (amandeepmittal) [inline docs/pages/accounts/sso.mdx:6]:
```suggestion
```

Not required to add a title since the first paragraph can serve as an intro to the whole page.

File: docs/pages/accounts/sso.mdx [hunk 2]
```diff
@@ -0,0 +1,62 @@
+---
+title: Single Sign-On (SSO)
+description: Learn about how your enterprise organization can use your Identity Provider to manage Expo users.
```
Comment (amandeepmittal) [inline docs/pages/accounts/sso.mdx:3]:
```suggestion
description: Learn about how your enterprise organization can use your Identity Provider to manage Expo users in your team.
```

I think we need to add context as to what "Expo users" is meant here.

File: docs/pages/accounts/sso.mdx [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+---
+
+## Configure SSO for your organization
+
+SSO is available for Enterprise Plan customers. To get started, contact us and we'll help you set it up for your organization.
```
Comment (amandeepmittal) [inline docs/pages/accounts/sso.mdx:8]:
```suggestion
SSO is available for [Enterprise Plan](https://expo.dev/pricing) customers. To get started, contact us and we'll help you set it up for your organization.
```

Adding a link to the pricing page.

File: docs/pages/accounts/sso.mdx [hunk 4]
```diff
@@ -0,0 +1,62 @@
...
+
+> The Expo SSO system should support any Identity Provider (IdP) that supports [OpenID Connect Discovery 1.0](https://openid.net/specs/openid-connect-discovery-1_0.html). We have tested [Okta](https://www.okta.com/) and plan to test others as requested.
```
Comment (amandeepmittal) [inline docs/pages/accounts/sso.mdx:10]:
```suggestion
> The Expo SSO system supports any Identity Provider (IdP) that supports [OpenID Connect Discovery 1.0](https://openid.net/specs/openid-connect-discovery-1_0.html). We have tested [Okta](https://www.okta.com/) and plan to test others as requested.
```

"should support" is not the phrase I'd like us to use in docs. It signifies that we are confident but not sure.

File: docs/pages/accounts/sso.mdx [hunk 5]
```diff
@@ -0,0 +1,62 @@
...
+
+## SSO user sign in
+
+### Expo website
+
+1. Navigate to https://expo.dev/sso-login and enter the account name of your organization.
```
Comment (amandeepmittal) [inline docs/pages/accounts/sso.mdx:16]:
```suggestion
1. Navigate to [expo.dev/sso-login](https://expo.dev/sso-login) and enter the account name of your organization.
```

Comment (amandeepmittal) [reply docs/pages/accounts/sso.mdx:16]: Btw, the link in above is 404. not sure if the page be up by the time docs are published.

File: docs/pages/accounts/sso.mdx [hunk 6]
```diff
@@ -0,0 +1,62 @@
...
+
+    > You can create a link that pre-fills the organization name. For example, https://expo.dev/sso-login/test-org will pre-fill `test-org`.
```
Comment (amandeepmittal) [inline docs/pages/accounts/sso.mdx:18]:
```suggestion
    > You can create a link that pre-fills the organization name. For example, [expo.dev/sso-login/test-org](https://expo.dev/sso-login/test-org) pre-fills `test-org`.
```

File: docs/pages/accounts/sso.mdx [hunk 7]
```diff
@@ -0,0 +1,62 @@
...
+
+1. Log in to your identity provider (IdP).
+1. You will then be asked to choose a Expo username. This will be your username in Expo.
```
Comment (amandeepmittal) [inline docs/pages/accounts/sso.mdx:21]:
```suggestion
1. Next, you'll be prompted to select an Expo username. This is your username in the Expo account.
```

File: docs/pages/accounts/sso.mdx [hunk 8]
```diff
@@ -0,0 +1,62 @@
+---
+title: Single Sign-On (SSO)
+description: Learn about how your enterprise organization can use your Identity Provider to manage Expo users.
+---
+
```
Comment (amandeepmittal) [inline docs/pages/accounts/sso.mdx:5]:
```suggestion

import { Terminal } from '~/ui/components/Snippet';
```

File: docs/pages/accounts/sso.mdx [hunk 9]
```diff
@@ -0,0 +1,62 @@
...
+1. Navigate to https://expo.dev/sso-login and enter the account name of your organization.
+
+    > You can create a link that pre-fills the organization name. For example, https://expo.dev/sso-login/test-org will pre-fill `test-org`.
+
+1. Log in to your identity provider (IdP).
+1. You will then be asked to choose a Expo username. This will be your username in Expo.
+
+### Expo CLI
+
+`npx expo login --sso`
```
Comment (amandeepmittal) [inline docs/pages/accounts/sso.mdx:25]:
```suggestion
When using Expo CLI, you can run the following command to log in to your Expo account.

<Terminal cmd={["$ npx expo login --sso"]} />
```

- Please add context as to when this command is useful or has to be used by the end user. I've added as a placeholder.
- Also use `Terminal` component to display a command.

File: docs/pages/accounts/sso.mdx [hunk 10]
```diff
@@ -0,0 +1,62 @@
...
+
+### EAS CLI
+
+`eas login --sso`
```
Comment (amandeepmittal) [inline docs/pages/accounts/sso.mdx:29]:
```suggestion
When using EAS CLI, you can run the following command to log in to your Expo account.

<Terminal cmd={["$ eas login --sso"]} />

```

File: docs/pages/accounts/sso.mdx [hunk 11]
```diff
@@ -0,0 +1,62 @@
...
+
+### Expo Go
+
+1. Go through normal sign in flow but click "Continue with SSO" button on sign in page.
```
Comment (amandeepmittal) [inline docs/pages/accounts/sso.mdx:33]:
```suggestion
1. Click the **Continue with SSO** button on the sign-in page when going through the sign-in flow.
```

I think we can put attention to the different step that user has to take when going through this step.

File: docs/pages/accounts/sso.mdx [hunk 12]
```diff
@@ -0,0 +1,62 @@
...
+1. Follow steps for Expo website above.
```
Comment (amandeepmittal) [inline docs/pages/accounts/sso.mdx:34]:
```suggestion
1. Follow the steps for the Expo website above.
```

Which steps from "above" do we refer to here?

File: docs/pages/accounts/sso.mdx [hunk 13]
```diff
@@ -0,0 +1,62 @@
...
+
+## SSO user restrictions
+
+SSO users are just like regular users for the most part, but there are a few known exceptions:
```
Comment (amandeepmittal) [inline docs/pages/accounts/sso.mdx:38]:
```suggestion
SSO users are like regular users. However, there are a few known exceptions:
```

We can remove "just" usage as it doesn't have significance here and make it more concise.

File: docs/pages/accounts/sso.mdx [hunk 14]
```diff
@@ -0,0 +1,62 @@
...
+- SSO users can only belong to their SSO organization. They also cannot create additional organizations.
+- SSO users cannot leave their SSO organization. Doing so deletes their SSO user.
+- SSO users cannot access the Expo forums.
+- SSO users cannot subscribe to EAS for their personal accounts.
+
+## SSO administration
+
+### Removing SSO users
```
Comment (amandeepmittal) [inline docs/pages/accounts/sso.mdx:46]:
```suggestion
### Remove SSO users
```

We've started using short form verbs instead of gerunds in section headings.

File: docs/pages/accounts/sso.mdx [hunk 15]
```diff
@@ -0,0 +1,62 @@
...
+
+If someone has left your organization, first remove/disable them in your IdP. This user will no longer have access to their Expo account in at most the token refresh duration you have configured in your IdP.
```
Comment (amandeepmittal) [inline docs/pages/accounts/sso.mdx:48]:
```suggestion
If someone has left your organization, remove or disable them in your IdP. This user will no longer have access to their Expo account in at most the token refresh duration you have configured in your IdP.
```

- Let's be a bit concise and authoritative.
- "their Expo account in at most the token refresh duration you have configured in your IdP" this phrase is confusing. as  a user I'm not sure I understand what is meant by "... at most the token refresh duration...". Does it mean by the time refresh token duration configured in their IdP expires?

File: docs/pages/accounts/sso.mdx [hunk 16]
```diff
@@ -0,0 +1,62 @@
...
+If you wish to remove them ahead of that time or you wish to remove them to clean up users on your account, you may do so in the organization member settings page.
```
Comment (amandeepmittal) [inline docs/pages/accounts/sso.mdx:49]:
```suggestion
If you wish to remove them ahead of that time or you wish to remove them to clean up users on your account, you may do so on the organization member settings page:
```

- Correct preposition usage.

File: docs/pages/accounts/sso.mdx [hunk 17]
```diff
@@ -0,0 +1,62 @@
...
+
+1. Navigate to your [organization account member settings](https://expo.dev/accounts/[account]/settings/members)
+2. Click the dropdown next to the member you wish to delete, and click "Delete SSO user"
```
Comment (amandeepmittal) [inline docs/pages/accounts/sso.mdx:52]:
```suggestion
2. Click the dropdown next to the member you wish to delete, and click **Delete SSO user**
```

File: docs/pages/accounts/sso.mdx [hunk 18]
```diff
@@ -0,0 +1,62 @@
...
+
+    > **warning** This will delete their personal account and all data associated with it. All data in your organization account will remain unaffected.
+
+### Changing billing or discontinuing use of SSO
```
Comment (amandeepmittal) [inline docs/pages/accounts/sso.mdx:56]:
```suggestion
### Change billing or discontinue use of SSO
```

File: docs/pages/accounts/sso.mdx [hunk 19]
```diff
@@ -0,0 +1,62 @@
...
+
+An active Enterprise Plan is required to continue using SSO. Contact us if you wish to discontinue use of SSO or change your plan.
```
Comment (amandeepmittal) [inline docs/pages/accounts/sso.mdx:58]:
```suggestion
An active Enterprise Plan is required to continue using SSO. [Contact us](https://expo.dev/contact) if you wish to discontinue the use of SSO or change your plan.
```

File: docs/pages/accounts/sso.mdx [hunk 20]
```diff
@@ -0,0 +1,62 @@
...
+
+### Deleting your SSO organization
```
Comment (amandeepmittal) [inline docs/pages/accounts/sso.mdx:60]:
```suggestion
### Delete SSO organization
```

File: docs/pages/accounts/sso.mdx [hunk 21]
```diff
@@ -0,0 +1,62 @@
...
+
+Once SSO is configured for an organization, account deletion must be done manually by Expo. Contact us for assistance.
```
Comment (amandeepmittal) [inline docs/pages/accounts/sso.mdx:62]:
```suggestion
Once SSO is configured for an organization, account deletion must be done manually by the Expo team. [Contact us](https://expo.dev/contact) for assistance.
```

File: docs/pages/accounts/sso.mdx (hunk 4 above)
Comment (keith-kurak) [reply docs/pages/accounts/sso.mdx:10]: Wonder if we could rephrase this to the effect of "We currently formally support Okta, but we implement the OIDC Discovery 1.0 spec, and are working to verify additional providers that implement the spec. If you use another provider and are interested in SSO, let us know."

File: docs/pages/accounts/sso.mdx (hunk 10 above)
Comment (keith-kurak) [inline docs/pages/accounts/sso.mdx:29]: For EAS and Expo CLI's, we should describe what happens, like, that your web browser will open and either take you the SSO login page and redirect, or redirect back immediately if you're already logged in.

File: docs/pages/accounts/sso.mdx [hunk 22]
```diff
@@ -0,0 +1,62 @@
...
+
+1. Go through normal sign in flow but click "Continue with SSO" button on sign in page.
+1. Follow steps for Expo website above.
+
+## SSO user restrictions
+
+SSO users are just like regular users for the most part, but there are a few known exceptions:
+- SSO users can only belong to their SSO organization. They also cannot create additional organizations.
+- SSO users cannot leave their SSO organization. Doing so deletes their SSO user.
+- SSO users cannot access the Expo forums.
```
Comment (keith-kurak) [inline docs/pages/accounts/sso.mdx:41]: Will it kick them out entirely or just not let them be logged in on the forums, comment, etc.?

**Review by keith-kurak — COMMENTED** [review]
Would it make sense to comment here to the new vs existing account support? E.g., 
- that we can support it for a new org right out of the gate
- we can add new SSO users to an existing previously-non-SSO org alongside the existing non-SSO users
- We do not support yet migrating existing non-SSO users to SSO, but are working on that.
(and maybe...)
- Existing users on Enterprise accounts can still start evaluating SSO by creating a new account. They can use their same email address for the SSO and non-SSO accounts, but will need a distinct username.

The more we document this here, the more we have for onboarding existing enterprise customers interested in SSO, and we temper any expectations that we're just going to be able to magically convert what's already there.

File: docs/pages/accounts/sso.mdx [hunk 23]
```diff
@@ -1,62 +1,72 @@
...
 - SSO users can only belong to their SSO organization. They also cannot create additional organizations.
 - SSO users cannot leave their SSO organization. Doing so deletes their SSO user.
-- SSO users cannot access the Expo forums.
+- SSO users cannot log in to the Expo forums.
 - SSO users cannot subscribe to EAS for their personal accounts.
 
 ## SSO administration
 
-### Removing SSO users
+Both new organizations and existing organizations are supported. Organizations with existing non-SSO users can enable SSO and both non-SSO users and SSO users can continue to be added. Migration of non-SSO users to SSO users is not currently supported.
```
Comment (keith-kurak) [inline docs/pages/accounts/sso.mdx:54]: I could see it the other way, but if it's the case that an SSO user can be added with the same email as a non-SSO user, it'd be my preference to document it. I feel like this is going to be the way existing orgs evaluate and/ or migrate to SSO initially. I can make it part of an internal script, but if it's part of the documentation, less need for someone to ask how they should migrate themselves and less room for me to second-guess myself here.

That's assuming this is what we would encourage them to do at all and don't feel like there's too much risk of confusion. I think it's worth a small-ish existing org to consider, because otherwise a key user can only get feedback about SSO by asking new org members how it's going (way slower feedback loop, less confidence in the offering).

Comment (wschurman) [reply docs/pages/accounts/sso.mdx:54]: I don't think we should encourage migration necessarily. Maybe it's a terminology thing. I guess I would describe it as "enabling SSO as a sign in option to the org" versus "migrating the org to SSO". So the person evaluating it would just be creating a new expo SSO user, but that's not what we should recommend for all their users to do.

By design, it is fine to have both SSO users and non-SSO users, and yes, they can have the same email. As for evaluation, I could see it as happening in the following order:
1. We set it up for them
2. Someone on their IT team tries it out with their account on their IdP (it can have the same email as their existing expo user account)
3. They decide to leave SSO enabled, and only new members are added via SSO. Old members can keep working as normal.

Comment (wschurman) [reply docs/pages/accounts/sso.mdx:54]: I'll document this a bit.

Comment (wschurman) [reply docs/pages/accounts/sso.mdx:54]: One other important thing is that SSO users don't know they have an email. It's an implementation detail, and not all SSO users do have an email either.

Comment (keith-kurak) [reply docs/pages/accounts/sso.mdx:54]: In the sort term I'm more focused on existing key users to be able to use SSO themselves, not necessarily migration. If I tell key user at partner X that SSO is ready for them, but it's just for new accounts, their first question is going to be, "well, how do I try it out?" Noting the second account path is possible covers that and at least hints at an informal migration strategy that could be pursued before we offer something more formal.

Even if it's not technically required, I think full migration/ adoption will be the goal for many orgs. Without full migration, these orgs do not get the full benefits of SSO (ability to revoke access, manage password/ MFA policies across, etc. for all users from their central identity provider).

**Review by keith-kurak — APPROVED** [review]

File: docs/pages/accounts/sso.mdx (hunk 23 above)
Comment (wschurman) [reply docs/pages/accounts/sso.mdx]: Yeah, that makes sense. I think we'll iron this out through the first couple onboardings and go from there. The preview release will definitely not be super smooth.
